### PR TITLE
[`ViTHybrid`] Fix `accelerate` slow tests

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

Fixes failing `ViTHybrid` `accelerate` tests

Uses the same procedure as DPTHybrid for `backbone_featmap_shape` - Now all slow tests should pass :-) 

cc @sgugger @ydshieh

### CODE REVIEW

File: src/transformers/models/vit_hybrid/modeling_vit_hybrid.py [hunk 1]
```diff
@@ -166,11 +166,10 @@ def __init__(self, config, feature_size=None):
         feature_dim = self.backbone.channels[-1]
 
         if feature_size is None:
-            dummy_image = torch.zeros(1, num_channels, image_size[0], image_size[1])
-            with torch.no_grad():
-                feature_map = self.backbone(dummy_image).feature_maps[-1]
```
Comment (younesbelkada) [inline src/transformers/models/vit_hybrid/modeling_vit_hybrid.py:171]: You predicted correctly the failing test in https://github.com/huggingface/transformers/pull/20649#pullrequestreview-1208498147 @sgugger !

**Review by sgugger — APPROVED** [review]
Thanks for the fix!

File: tests/models/vit_hybrid/test_modeling_vit_hybrid.py [hunk 2]
```diff
@@ -57,6 +57,7 @@ def __init__(
         attention_probs_dropout_prob=0.1,
         type_sequence_label_size=10,
         initializer_range=0.02,
+        backbone_featmap_shape=[1, 1024, 4, 4],
```
Comment (ydshieh) [inline tests/models/vit_hybrid/test_modeling_vit_hybrid.py:60]: I guess `4` is from the fact we have `image_size=64` here in the test, right?
Is `1024` necessary, or we can change it to something smaller?

**Review by ydshieh — APPROVED** [review]
Thanks! Have a question regarding `1024` -> would be great if we can use a smaller value.

File: tests/models/vit_hybrid/test_modeling_vit_hybrid.py (hunk 2 above)
Comment (younesbelkada) [reply tests/models/vit_hybrid/test_modeling_vit_hybrid.py:60]: Probably you are right here, let me double check

Comment (younesbelkada) [reply tests/models/vit_hybrid/test_modeling_vit_hybrid.py]: Yes we can! Fixed in [f24a507](https://github.com/huggingface/transformers/pull/20679/commits/f24a507a5bf6c401ab2379d3ed73a8e96ef32f50)

File: src/transformers/models/vit_hybrid/configuration_vit_hybrid.py [hunk 3]
```diff
@@ -103,6 +105,7 @@ def __init__(
         image_size=224,
         patch_size=1,
         num_channels=3,
+        backbone_featmap_shape=[1, 1024, 24, 24],
```
[the remark this answers is not in the document]
Comment (younesbelkada) [reply src/transformers/models/vit_hybrid/configuration_vit_hybrid.py:108]: Let's maybe leave it as it is now, and having it auto-computed later on!

File: tests/models/vit_hybrid/test_modeling_vit_hybrid.py (hunk 2 above)
Comment (younesbelkada) [reply tests/models/vit_hybrid/test_modeling_vit_hybrid.py]: Smaller values in [0b28767](https://github.com/huggingface/transformers/pull/20679/commits/0b28767e70978ec4076fb1523f241973a743aa43)

**Pull request merged**